src/terrain/draw_terrain_raster.js

```diff
@@ -141,11 +142,13 @@ function drawTerrainForGlobe(painter: Painter, terrain: Terrain, sourceCache: So
 
     let program, programMode;
     const showWireframe = painter.options.showTerrainWireframe ? SHADER_TERRAIN_WIREFRAME : SHADER_DEFAULT;
+    const tr = painter.transform;
+    const useCustomAntialiasing = !painter.style.map._antialias && !!context.extStandardDerivative && globeToMercatorTransition(tr.zoom) === 0.0;
 
     const setShaderMode = (mode, isWireframe) => {
         if (programMode === mode)
             return;
-        const defines = [shaderDefines[mode], 'PROJECTION_GLOBE_VIEW'];
+        const defines = useCustomAntialiasing ? [shaderDefines[mode], 'CUSTOM_ANTIALIASING', 'PROJECTION_GLOBE_VIEW'] : [shaderDefines[mode], 'PROJECTION_GLOBE_VIEW'];
```

Comment:
This is a bit hard to read on one line, just for the sake of splitting these statements a bit more and have it less cluttered, how about following the pattern that's already there with `isWireframe`:

```suggestion
        const defines = [shaderDefines[mode], 'PROJECTION_GLOBE_VIEW'];
        if (useCustomAntialiasing) {
            defines.push('CUSTOM_ANTIALIASING');
        }
```